src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -45,12 +52,15 @@
     @NonNull
     private final Secret privateKey;
 
+    // Refresh the token if it is near expiration
+    private final long tokenExpirationBufferSeconds = Duration.ofMinutes(5).getSeconds();
```

Comment:
this doesn't look like it needs saving to disk ?
```suggestion
    private transient final long tokenExpirationBufferSeconds = Duration.ofMinutes(5).getSeconds();
```

Reply:
ah right it's final so no need for this

Reply:
Oops, I meant for it to be `static`.  Will change.